app/src/hid_keyboard.c

```diff
@@ -0,0 +1,347 @@
+#include "util/log.h"
+#include "hid_keyboard.h"
+
+/**
+ * For HID over AOAv2, we only need report descriptor.
+ * Normally a basic HID keyboard uses 8 bytes,
+ * `Modifier Reserved Key Key Key Key Key Key`.
+ * See Appendix B.1 Protocol 1 (Keyboard) and
+ * Appendix C: Keyboard Implementation in
+ * <https://www.usb.org/sites/default/files/hid1_11.pdf>.
+ * But if we want to support media keys on keyboard,
+ * we need to use two reports,
+ * report id 1 and usage page key codes for basic keyboard,
+ * report id 2 and usage page consumer for media keys.
+ * See 8. Report Protocol in
+ * <https://www.usb.org/sites/default/files/hid1_11.pdf>.
+ * The former byte is item type prefix, we only use short items here.
+ * For how to calculate an item, read 6.2.2 Report Descriptor in
+ * <https://www.usb.org/sites/default/files/hid1_11.pdf>.
+ * For Consumer Page tags, see 15 Consumer Page (0x0C) in
+ * <https://usb.org/sites/default/files/hut1_2.pdf>.
+ */
+/**
+ * You can dump your device's report descriptor with
+ * `sudo usbhid-dump -m vid:pid -e descriptor`.
+ * Change `vid:pid` to your device's vendor ID and product ID.
+ */
+unsigned char kb_report_desc_buffer[]  = {
+    // Usage Page (Generic Desktop)
+    0x05, 0x01,
+    // Usage (Keyboard)
+    0x09, 0x06,
+
+    // Collection (Application)
+    0xA1, 0x01,
+    // Report ID (1)
+    0x85, 0x01,
+
+    // Usage Page (Keyboard)
+    0x05, 0x07,
+    // Usage Minimum (224)
+    0x19, 0xE0,
+     // Usage Maximum (231)
+    0x29, 0xE7,
+    // Logical Minimum (0)
+    0x15, 0x00,
+    // Logical Maximum (1)
+    0x25, 0x01,
+    // Report Size (1)
+    0x75, 0x01,
+    // Report Count (8)
+    0x95, 0x08,
+    // Input (Data, Variable, Absolute): Modifier byte
+    0x81, 0x02,
+
+    // Report Size (8)
+    0x75, 0x08,
+    // Report Count (1)
+    0x95, 0x01,
+    // Input (Constant): Reserved byte
+    0x81, 0x01,
```

Comment:
Is it necessary (it works without)?

Reply:
Every examples on Internet have those reserved byte, so we'd better keep it.